server/src/main/java/io/crate/session/Session.java

```diff
@@ -164,6 +165,22 @@ public class Session implements AutoCloseable {
     private TransactionState currentTransactionState = TransactionState.IDLE;
     private volatile String lastStmt;
 
+    Session(Session session) {
+        this.id = session.id;
+        this.connectionProperties = session.connectionProperties;
+        this.timeCreated = session.timeCreated;
+        this.secret = session.secret;
+        this.analyzer = session.analyzer;
+        this.planner = session.planner;
+        this.jobsLogs = session.jobsLogs;
+        this.isReadOnly = session.isReadOnly;
+        this.executor = session.executor;
+        this.sessionSettings = session.sessionSettings;
+        this.onClose = session.onClose;
+        this.tempErrorRetryCount = session.tempErrorRetryCount;
+
+    }
```

Comment:
This and `protected long phaseTime(String phase, long phaseStart)` are test related changes - didn't find a better way to test long parse/analysis phases.